### PR TITLE
Added the Chrono Legionnaire

### PR DESCRIPTION
**Pull request by Mailaender**
It is bit in placeholder quality due to the exotic weaponry. ~~Not sure why I can't get it to shoot at all though.~~

### CODE REVIEW

**Comment by abcdefg30** [comment]
We're still missing: `ProducibleWithLevel` and the offset of the selection bounds (?) is wrong.

File: weapons/allied-infantry.yaml [hunk 1]
```diff
@@ -294,3 +294,37 @@ awp:
...
+		Duration: 10
+		DamageInterval: 5
+		Width: 0c512
+		Shape: Flat
+		ZOffset: 2047
+		BeyondTargetRange: 0c256
+		Blockable: true
+		Color: 00FFFFFF
+		ValidTargets: Ground
+		Versus:
```
Comment (MustaphaTR) [inline weapons/allied-infantry.yaml]: Why versus vaule is on Projictle?

That's possibly why it doesn't fire.

**Comment by Mailaender** [comment]
Updated. It now fires and has has selection bounds. Thanks guys.

File: weapons/allied-infantry.yaml [hunk 2]
```diff
@@ -294,3 +294,33 @@ awp:
...
+	Report: ichratta.wav
+	Projectile: LaserZap
+		BeamWidth: 2
+		BeamDuration: 7
+		Color: 00FFFFFF
+	Warhead@Damage: SpreadDamage
+		Spread: 64
+		Damage: 8
+		Versus:
+			None: 100
```
Comment (GraionDilach) [inline weapons/allied-infantry.yaml]: You don't need these, 100 is default.

File: weapons/allied-infantry.yaml [hunk 3]
```diff
@@ -294,3 +294,33 @@ awp:
...
 		Explosions: piff
 		ValidImpactTypes: Water
+
+NeutronRifle:
+	ValidTargets: Ground
+	ReloadDelay: 120
+	Range: 10c0
+	MinRange: 0c001
+	Report: ichratta.wav
+	Projectile: LaserZap
```
Comment (GraionDilach) [inline weapons/allied-infantry.yaml]: A TeslaZap DimZap: 1 BrightZap :0 using a custom RGBA palette is closer what RA2 had imo.

**Comment by GraionDilach** [comment]
Will do as a start though. :+1: when the warhead versus fixup is done, changing the weapon type again is optional.

**Comment by Mailaender** [comment]
Updated.

**Comment by MustaphaTR** [comment]
Looks like needs a rebase.

**Comment by Mailaender** [comment]
Rebased.

File: rules/allied-infantry.yaml [hunk 4]
```diff
@@ -382,3 +382,41 @@ jumpjet:
...
+		PipType: Red
+	-Crushable:
+	PortableChrono:
+		ChronoshiftSound: ichrmova.wav
+		DeployCursor: chronosphere
+		DeployBlockedCursor: chronosphere
+		Voice: Move
+	Voiced:
+		VoiceSet: ChronoLegionnaireVoice
+		ChargeDelay: 300
```
Comment (abcdefg30) [inline rules/allied-infantry.yaml]: This doesn't look correct.

File: weapons/allied-infantry.yaml [hunk 5]
```diff
@@ -173,3 +173,22 @@ AWPE:
 20mme:
 	Inherits: 20mm
 	Burst: 2
+
+NeutronRifle:
+	ValidTargets: Ground
+	ReloadDelay: 120
+	Range: 10c0
+	MinRange: 0c001
```
Comment (abcdefg30) [inline weapons/allied-infantry.yaml]: Where does this come from?

Comment (Mailaender) [inline weapons/allied-infantry.yaml]: https://github.com/OpenRA/ra2/blob/70f14b00d3b3f36e8339255c72d9811610444181/weapons/allied-infantry.yaml#L150 I assume.

Comment (GraionDilach) [inline weapons/allied-infantry.yaml]: That shouldn't be needed here - jumpjet stuff can crash at range 0 for some reason. CLeg ain't jumpjet.

File: sequences/allied-infantry.yaml [hunk 6]
```diff
@@ -480,3 +479,58 @@ jumpjet:
...
+		Facings: 1
+	die1:
+		Start: 38
+		Length: 15
+		Facings: 8
+	die2:
+		Start: 53
+		Length: 15
+		Facings: 8
+	die3:
```
Comment (GraionDilach) [inline sequences/allied-infantry.yaml]: These and onwards need an update.

**Comment by Mailaender** [comment]
Updated.

File: rules/defaults.yaml [hunk 7]
```diff
@@ -62,6 +62,17 @@
 	TimedUpgradeBar:
 		Upgrade: invulnerability
 
+^ChronoDisable:
```
Comment (MustaphaTR) [inline rules/defaults.yaml]: You should make that effect production, sws, weapons and stuff like that, they would be abit more useful for now. Also the Alpha effect can be added. It must be already possible.

File: rules/allied-infantry.yaml [hunk 8]
```diff
@@ -382,3 +382,41 @@ jumpjet:
...
+	Inherits: ^Infantry
+	Selectable:
+		Bounds: 20, 30, 0, -11
+	Tooltip:
+		Name: Chrono Legionnaire
+		Description: High-tech soldier.\n  Strong vs Infantry, Vehicles\n  Weak vs Aircraft
+	Buildable:
+		Queue: Infantry
+		BuildAtProductionType: Infantry
+		BuildPaletteOrder: 60
```
Comment (MustaphaTR) [inline rules/allied-infantry.yaml]: I think this is too low. This must be go to last, as it has TechLevel=10 in original game.

File: rules/defaults.yaml (hunk 7 above)
Comment (GraionDilach) [inline rules/defaults.yaml]: Please read #174.

**Pull request closed without merging**